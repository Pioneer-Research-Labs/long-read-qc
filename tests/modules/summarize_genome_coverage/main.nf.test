nextflow_process {

    name "Test Process summarize_genome_coverage"
    script "modules/summarize_genome_coverage/main.nf"
    process "summarize_genome_coverage"

    test("Should generate a graphic of genome coverage") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                 input[0] = file('${projectDir}/test_data/genome_coverage.tsv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/genome_coverage.png").match()
        }

    }

}
