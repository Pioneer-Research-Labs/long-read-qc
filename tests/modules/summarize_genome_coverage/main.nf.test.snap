{
    "Should generate a graphic of genome coverage": {
        "content": [
            {
                "0": [
                    "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/d6bf9103d82a91968d05caec7fc06086/work/02/99a4404f481d7b7cc7f3c2f4c2c007/genome_coverage.png"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-12T17:37:06.418505"
    }
}